emscripten.py

```diff
@@ -582,10 +582,13 @@ def read_proxied_function_signatures(asmConsts):
 
 
 def compile_settings(compiler_engine, libraries, temp_files):
+  settings = shared.Settings.to_dict()
+  settings.update((key, int(value)) for key, value in settings.items() if isinstance(value, bool))
+
```

Comment:
these changes also seem like they should be in separate PRs?

Reply:
Yes, it's part of #7105